cmd/syft/cli/commands.go

```diff
@@ -55,6 +55,7 @@ func New() (*cobra.Command, error) {
 
 	// rootCmd is currently an alias for the packages command
 	rootCmd := &cobra.Command{
+		Use:           fmt.Sprintf("%s package [SOURCE]", internal.ApplicationName),
```

Comment:
`package` is not a valid command, and this is the root command, so this should probably be:
```go
fmt.Sprintf("%s [SOURCE]", internal.ApplicationName)
```

Reply:
Even so this is the _root_ command, not the packages command (even though they are duplicates), so it should display usage text as the root command.

Reply:
Using `fmt.Sprintf("%s packages [SOURCE]",...` causes the following to be output for the 'Usage'

```
$ syft
Generate a packaged-based Software Bill Of Materials (SBOM) from container images and filesystems

Usage:
  syft packages [SOURCE] [flags]
  syft [command]

Examples:
  syft packages alpin...
```

Happy to change that to what every you prefer. The important bit (for our fix) is that we have `syft` at the start of the command. That is what the `spf13/cobra` library needs to generate a working shell completion script (for bash at least!!)

Reply:
Right, but since a user can run syft without the `packages` command, I think we should leave this as `"%s [SOURCE]"`

Reply:
Hmmm... So the `[SOURCE]` bit of that usage command is only really relevant when running `syft packages ...`

```
$ syft packages
Generate a packaged-based Software Bill Of Materials (SBOM) from container images and filesystems

Usage:
  syft packages [SOURCE] [flags]
```

Perhaps we should just have `fmt.Sprintf("%s [command] [flags]", ...` as the usage for the main syft command?

This would just give:

```
$ syft
Generate a packaged-based Software Bill Of Materials (SBOM) from container images and filesystems

Usage:
  syft [command] [flags]
  syft [command]

Examples:
  syft packages alpine:latest                                a summary of discovered packages
  syft packages alpine:la...
```

...which seems to make a bit more sense. Are you OK with that?

Reply:
There's a bit of a mismatch between the help text and the command itself (this should be fixed, but I would not ask you to change this as part of this PR).

You can run Syft like:

```
syft alpine:latest
```

_without_ the `packages` command keyword, because the root command duplicates the functionality of the packages command.

For this reason, I would prefer that the root help text indicates how to run the root command, rather than indicating `packages` or some sub-command is necessary. It's a very small detail.

Reply:
Ahh... OK. That makes sense now. The penny finally dropped!! :smile: